src/utilities.cpp

```diff
@@ -45,6 +46,24 @@ Eigen::VectorXd DiscreteDifferentiation(const Eigen::VectorXd& input_vector, dou
   return derivative;
 };
 
+Eigen::VectorXd DiscreteDifferentiationWithFiltering(const Eigen::VectorXd& input_vector, const double timestep,
+                                                     const double first_element, const double filter_coefficient)
+{
+  Eigen::VectorXd derivative = DiscreteDifferentiation(input_vector, timestep, first_element);
+
+  // Apply a low-pass filter
+  ButterworthFilter filter(filter_coefficient);
+  // Set the initial value
+  filter.reset(derivative(0));
+  for (size_t point = 1; point < derivative.size(); ++point)
+  {
+    // Lowpass filter the position command
+    derivative(point) = filter.filter(derivative(point));
```

Comment:
Do we care about phase shift? If so, it's a common practice to filter a signal forwards, then filter it backwards to remove any phase shift. If we don't care, then it doesn't really matter though.

Reply:
Hm, I didn't know that but it makes sense. That seems like it could be important, huh? Want to implement it and push to this branch?

Reply:
OK, I've pushed it. :+1: